crates/router/src/core/payments/operations/payment_create_intent.rs

```diff
@@ -0,0 +1,280 @@
+            hyperswitch_domain_models::payments::PaymentIntent::create_domain_model_from_request(
+                state.into(),
+                key_store,
+                payment_id,
+                merchant_account,
+                profile,
+                request.clone(),
+            )
+            .await?;
+
+        let payment_intent = db
+            .insert_payment_intent(
+                key_manager_state,
+                payment_intent_domain,
+                key_store,
+                storage_scheme,
+            )
+            .await
+            .to_duplicate_response(errors::ApiErrorResponse::GenericDuplicateError {
+                message: format!(
+                    "Payment Intent with payment_id {} already exists",
+                    payment_id.get_string_repr()
+                ),
+            })
+            .attach_printable("failed while inserting new payment intent")?;
+
+        let payment_data = payments::PaymentIntentData {
+            flow: PhantomData,
+            payment_intent,
+        };
+
+        let get_trackers_response = operations::GetTrackerResponse {
+            operation: Box::new(self),
+            payment_data,
+        };
+
+        Ok(get_trackers_response)
+    }
+}
+
+#[async_trait]
+impl<F: Clone> UpdateTracker<F, payments::PaymentIntentData<F>, PaymentsCreateIntentRequest>
+    for PaymentCreateIntent
+{
+    #[instrument(skip_all)]
+    async fn update_trackers<'b>(
+        &'b self,
+        state: &'b SessionState,
+        _req_state: ReqState,
+        payment_data: payments::PaymentIntentData<F>,
+        _customer: Option<domain::Customer>,
+        storage_scheme: enums::MerchantStorageScheme,
+        _updated_customer: Option<storage::CustomerUpdate>,
+        key_store: &domain::MerchantKeyStore,
+        _frm_suggestion: Option<FrmSuggestion>,
+        _header_payload: api::HeaderPayload,
+    ) -> RouterResult<(
+        PaymentsCreateIntentOperation<'b, F>,
+        payments::PaymentIntentData<F>,
+    )>
+    where
+        F: 'b + Send,
+    {
+        Ok((Box::new(self), payment_data))
+    }
+}
+
+impl<F: Send + Clone>
+    ValidateRequest<F, PaymentsCreateIntentRequest, payments::PaymentIntentData<F>>
+    for PaymentCreateIntent
+{
+    #[instrument(skip_all)]
+    fn validate_request<'a, 'b>(
+        &'b self,
+        _request: &PaymentsCreateIntentRequest,
+        merchant_account: &'a domain::MerchantAccount,
+    ) -> RouterResult<(
+        PaymentsCreateIntentOperation<'b, F>,
+        operations::ValidateResult,
+    )> {
+        Ok((
+            Box::new(self),
+            operations::ValidateResult {
+                merchant_id: merchant_account.get_id().to_owned(),
+                storage_scheme: merchant_account.storage_scheme,
+                requeue: false,
+            },
+        ))
+    }
+}
+
+#[async_trait]
+impl<F: Clone + Send> Domain<F, PaymentsCreateIntentRequest, payments::PaymentIntentData<F>>
```

Comment:
@jarnura we do not need to get the customer details, we just have to validate if the customer_id exists and belongs to this merchant. Can we instead move this to get trackers?

Reply:
We don't want this to move into get_trackers for now, when we need to reduce the latency we can think about moving this to get_trackers or making concurrent calls with seperate domain and get_trackers